### DISCUSSION TITLE
Refactor Post Filtering

### DISCUSSION THREAD

**Discussion by onkoe** [Docile-Alligator/Infinity-For-Reddit#987] [Ideas]
The Boost implementation for post filtering is much simpler. You just check some boxes depending on your wishes. I would prefer an implementation of this with an "advanced"/"presets" option to access the current system.

Here's a comparison of each workflow:

## Boost
To filter posts, you [hit the three dots menu](https://user-images.githubusercontent.com/66580279/161599845-9ab43aad-3698-4fb7-bb49-eab290a80782.png), then [head to Content filters](https://user-images.githubusercontent.com/66580279/161599961-2aaaf122-f5e6-43fb-9c16-f658044e902b.png). There are session-based filters in the "simple" version of the menu, meaning that they are quick and temporary. If you wish to access more advanced filtering options, [you just hit More](https://user-images.githubusercontent.com/66580279/161600190-8b4b5e20-b3fa-4b48-8c4f-3b07b42d8c91.png).

## Infinity
For Infinity, everything is a lot more complex. You follow the same steps to reach the Filtering menu, but find that [you are in a new screen](https://user-images.githubusercontent.com/66580279/161600618-e39b50de-bfe9-48ff-90fc-f9a40361b61d.png) of filtered posts. Already, this is a rather odd interaction. If you are filtering an original set of posts, it's confusing to have a copy of the original. 

Anyways, once you reach the actual filtering menu, it's a bit [unclear what everything does](https://user-images.githubusercontent.com/66580279/161600883-5d5dce80-8fae-4957-9e70-1ae237ff52b3.png). It's considerably more time-consuming. Post filters often don't work, especially for videos only (where they typically count as links).

A refactor of this workflow may greatly enhance the power user experience with Infinity. I like to typically hang around and answer tech support text threads - Infinity is constantly worse on the filtering side. Any feedback, plans, or criticism would be greatly appreciated!

**Comment by mcstafford-git** [comment]
There is VERY flexible, utilitarian functionality available in filtering, but everything else in the app feels as though more attention to UI ease of use has been invested.

It would be nice to have a one-click "add sub to filter" option.
